streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedKCogroupedStreamImpl.java

```diff
@@ -0,0 +1,136 @@
+                                     final Map<KGroupedStreamImpl<K, T>, Aggregator<? super K, ? super T, V>> groupPatterns) {
+        super(name, keySerde, valSerde, sourceNodes, streamsGraphNode, builder);
+        this.sessionWindows = sessionWindows;
+        this.aggregateBuilder = aggregateBuilder;
+        this.groupPatterns = groupPatterns;
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Merger<? super K, V> sessionMerger,
+                                            final Materialized<K, V, SessionStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(initializer, "initializer can't be null");
+        Objects.requireNonNull(sessionMerger, "sessionMerger can't be null");
+        Objects.requireNonNull(materialized, "materialized can't be null");
+        final MaterializedInternal<K, V, SessionStore<Bytes, byte[]>> materializedInternal = new MaterializedInternal<>(
+            materialized,
+            builder,
+            AGGREGATE_NAME);
+        return aggregateBuilder.build(groupPatterns,
+                                      initializer,
+                                      NamedInternal.empty(),
+                                      materialize(materializedInternal),
+                                      materializedInternal.keySerde()
+                                          != null
+                                          ? new WindowedSerdes.SessionWindowedSerde<>(
+                                          materializedInternal
+                                              .keySerde()) : null,
+                                      materializedInternal.valueSerde(),
+                                      null,
+                                      sessionWindows,
+                                      sessionMerger);
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Merger<? super K, V> sessionMerger) {
+        return aggregate(initializer, sessionMerger, Materialized.with(keySerde, null));
+    }
+
+    @SuppressWarnings("deprecation") // continuing to support SessionWindows#maintainMs in fallback mode
+    private  StoreBuilder<SessionStore<K, V>> materialize(final MaterializedInternal<K, V, SessionStore<Bytes, byte[]>> materialized) {
+        SessionBytesStoreSupplier supplier = (SessionBytesStoreSupplier) materialized.storeSupplier();
+        if (supplier == null) {
+            // NOTE: in the future, when we remove sessionWindows#maintainMs(), we should set the default retention
+            // to be (sessionWindows.inactivityGap() + sessionWindows.grace()). This will yield the same default behavior.
+            final long retentionPeriod = materialized.retention() != null ? materialized.retention().toMillis() : sessionWindows.maintainMs();
```

Comment:
Wondering if we can get rid of supporting deprecated API for this case? \cc @vvcephei WDYT?